### DISCUSSION TITLE
How can I get the email from a token?

### DISCUSSION THREAD

**Discussion by renatocaliari** [pockethost/pockethost#221] [Q&A] [answered]
Apparently the URL "auth/confirm-password-reset" is able to retrieve the email from a token. 
As I am developing my own password confirmation screen, I would like to know how to do the same to display it on the screen.

Thanks.

**Comment by benallfree** [comment]
@renatocaliari PocketHost itself uses a custom confirmation screen. Our email template in the admin uses `https://pockethost.io/login/password-reset/confirm?token={TOKEN}`.

Check out https://github.com/benallfree/pockethost/tree/master/packages/pockethost.io/src/routes/login/password-reset to see how we do the frontend.

https://github.com/benallfree/pockethost/blob/master/packages/pockethost.io/src/routes/login/password-reset/confirm/%2Bpage.svelte#L22 shows how the UI works.

https://github.com/benallfree/pockethost/blob/master/packages/pockethost.io/src/pocketbase/PocketbaseClient.ts#L98 shows how to make the actual call to PocketBase via the JS SDK:

```ts
client
        .collection('users')
        .confirmPasswordReset(token, password, password)
```

I hope that helps!

**Reply by renatocaliari** [reply]
Thank you for the help, @benallfree.

I'm using the `confirmPasswordReset` method.

My question is regarding the screen that the default "Password Reset" email links to (`https://{myinstance}.pockethost.io/_/#/auth/confirm-password-reset`). This URL opens a screen that displays the email for which the token was generated to create a new password. 
![Screenshot 2023-06-12 12 02 47](https://github.com/benallfree/pockethost/assets/1719651/717c982d-b873-4c2b-9d97-dee6ae206761)

I'm not sure how I can do this to display the email on my screen as well. 

From what I noticed, it seems that the page `packages/pockethost.io/src/routes/login/password-reset/confirm/+page.svelte` also doesn't display the email. 

Thank you very much.

**Reply by benallfree** [reply]
I understand now. Decoding that token is probably a question for the PocketBase forum. Have you tried a simple base64 decode to see what the token contains?

@ganigeorgiev would certainly know this answer :)

**Reply by benallfree** [reply]
This might be a clue https://github.com/pocketbase/pocketbase/blob/d1bcaa65b125f642848113fea8c4a7cba378007f/ui/src/components/admins/PageAdminConfirmPasswordReset.svelte#L15

**Reply by ganigeorgiev** [reply, marked as answer]
Yes, the email is available if you base64 decode the token payload. If you are using the JS SDK, there is even an exposed helper that you can use:

```js
import PocketBase, { getTokenPayload } from "pocketbase";

...

const params = new URLSearchParams(window.location.search);

const email = getTokenPayload(params.get('token')).email;
```

**Reply by renatocaliari** [reply]
Thank you @benallfree  and @ganigeorgiev .

`getTokenPayload` worked perfectly for what I wanted.
